CRM/Volunteer/Form/Defaults.php

```diff
@@ -114,18 +106,29 @@ function validate() {
   function postProcess() {
     $values = $this->exportValues();
 
-    CRM_Core_BAO_Setting::setItem($values['volunteer_default_profiles_individual'],"volunteer_defaults", "volunteer_default_profiles_individual");
-    CRM_Core_BAO_Setting::setItem($values['volunteer_default_profiles_group'],"volunteer_defaults", "volunteer_default_profiles_group");
-    CRM_Core_BAO_Setting::setItem($values['volunteer_default_profiles_both'],"volunteer_defaults", "volunteer_default_profiles_both");
-    CRM_Core_BAO_Setting::setItem($values['volunteer_default_campaign'],"volunteer_defaults", "volunteer_default_campaign");
-    CRM_Core_BAO_Setting::setItem($values['volunteer_default_locblock'],"volunteer_defaults", "volunteer_default_locblock");
 
-    $isActive = array_key_exists("volunteer_default_is_active", $values) ? $values['volunteer_default_is_active'] : 0;
-    CRM_Core_BAO_Setting::setItem($isActive, "volunteer_defaults", "volunteer_default_is_active");
+    //Compose the profiles before we save tem.
+    $profiles = array();
+
+    foreach(CRM_Volunteer_BAO_Project::getProjectProfileAudienceTypes() as $audience) {
+      $profiles[$audience['type']] = CRM_Utils_Array::value('volunteer_default_profiles_' . $audience['type'], $values);
```

Comment:
I don't think it matters. I've tested it with single, multiple and null values and they all work fine. This only stores the setting, and when we parse the setting later we are looking for values, and only populate based on the presence of those values.